src/patternfly/components/ContextSelector/context-selector.scss

```diff
@@ -0,0 +1,181 @@
+
+  // Menu Menu Item
+  --pf-c-context-selector__menu-menu--MaxHeight: #{pf-size-prem(200px)};
+
+  // Menu Item
+  --pf-c-context-selector__menu-menu-item--PaddingTop: var(--pf-global--spacer--sm);
+  --pf-c-context-selector__menu-menu-item--PaddingRight: var(--pf-global--spacer--lg);
+  --pf-c-context-selector__menu-menu-item--PaddingBottom: var(--pf-global--spacer--sm);
+  --pf-c-context-selector__menu-menu-item--PaddingLeft: var(--pf-global--spacer--lg);
+  --pf-c-context-selector__menu-menu-item--hover--BackgroundColor: var(--pf-global--BackgroundColor--light-300);
+  --pf-c-context-selector__menu-menu-item--disabled--Color: var(--pf-global--Color--dark-200);
+  --pf-c-context-selector__menu-menu-item--disabled--BackgroundColor: transparent;
+
+
+  position: relative;
+  display: inline-block;
+  width: var(--pf-c-context-selector--Width);
+  max-width: 100%;
+}
+
+.pf-c-context-selector__toggle {
+  position: relative;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  min-width: var(--pf-c-context-selector__toggle--MinWidth);
+  min-width: 100%;
+  padding: var(--pf-c-context-selector__toggle--PaddingTop) var(--pf-c-context-selector__toggle--PaddingRight) var(--pf-c-context-selector__toggle--PaddingBottom) var(--pf-c-context-selector__toggle--PaddingLeft);
+  color: var(--pf-c-context-selector__toggle--Color);
+  white-space: nowrap;
+  cursor: pointer;
+  background-color: var(--pf-c-context-selector__toggle--BackgroundColor);
+  border: none;
+
+  & > * {
+    color: var(--pf-c-context-selector__toggle--Color);
+  }
+
+  &::before {
+    position: absolute;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+    content: "";
+    border: var(--pf-c-context-selector__toggle--BorderWidth) solid;
+    border-color: var(--pf-c-context-selector__toggle--BorderTopColor) var(--pf-c-context-selector__toggle--BorderRightColor) var(--pf-c-context-selector__toggle--BorderBottomColor) var(--pf-c-context-selector__toggle--BorderLeftColor);
+  }
+
+  &:hover,
+  &.pf-m-hover {
+    &::before {
+      --pf-c-context-selector__toggle--BorderBottomColor: var(--pf-c-context-selector__toggle--hover--BorderBottomColor);
+    }
+  }
+
+  &:active,
+  &.pf-m-active,
+  &:focus-within {
+    &::before {
+      --pf-c-context-selector__toggle--BorderBottomColor: var(--pf-c-context-selector__toggle--active--BorderBottomColor);
+
+      border-bottom-width: var(--pf-c-context-selector__toggle--active--BorderBottomWidth);
+    }
+  }
+
+  .pf-m-expanded > & {
+    &::before {
+      --pf-c-context-selector__toggle--BorderBottomColor: var(--pf-c-context-selector__toggle--expanded--BorderBottomColor);
+
+      border-bottom-width: var(--pf-c-context-selector__toggle--expanded--BorderBottomWidth);
+    }
+  }
+
+  .pf-c-context-selector__toggle-icon {
+    margin-left: var(--pf-c-context-selector__toggle-icon--MarginLeft);
+  }
+
+  .pf-c-context-selector__toggle-text {
```

Comment:
if you put a ton of text in here, it just makes the button wider and doesn't truncate.

Reply:
awesome added a max-width